chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -317,30 +317,23 @@ class ChainHandler(
 
   /** @inheritdoc */
   override def nextFilterHeaderBatchRange(
-      prevStopHash: DoubleSha256DigestBE,
+      startHeight: Int,
       batchSize: Int): Future[Option[FilterSyncMarker]] = {
-    val startHeightF = if (prevStopHash == DoubleSha256DigestBE.empty) {
-      Future.successful(0)
-    } else {
-      for {
-        prevStopHeaderOpt <- getFilterHeader(prevStopHash)
-        prevStopHeader = prevStopHeaderOpt.getOrElse(
-          throw UnknownBlockHash(s"Unknown block hash ${prevStopHash}"))
-      } yield prevStopHeader.height + 1
-    }
+    val stopHeight = startHeight - 1 + batchSize
 
-    for {
-      startHeight <- startHeightF
-      filterHeaderCount <- getFilterHeaderCount()
-      stopHeight =
-        if (startHeight - 1 + batchSize > filterHeaderCount)
-          filterHeaderCount
-        else startHeight - 1 + batchSize
-      stopBlockOpt <- getFilterHeadersAtHeight(stopHeight).map(_.headOption)
-      stopBlock = stopBlockOpt.getOrElse(
-        throw UnknownBlockHeight(s"Unknown filter header height ${stopHeight}"))
-    } yield {
-      if (startHeight > stopHeight)
+    val stopBlockF =
+      getFilterHeadersAtHeight(stopHeight).map(_.headOption).flatMap {
```

Comment:
Why `.map(_.headOption)` here? This makes us lose reorg safety

Reply:
That is what was being doine before.